contracts/globalConstraints/EtherGC.sol

```diff
@@ -0,0 +1,84 @@
+pragma solidity 0.5.17;
+
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "./GlobalConstraintInterface.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+
+/**
+ * @title EtherGC ether constraint per period
+ */
+contract EtherGC {
+    using SafeMath for uint256;
+
+    uint256 public constraintPeriodWindow; //a static window
+    uint256 public amountAllowedPerPeriod;
+    address public avatar;
+    uint256 public latestEthBalance;
+    mapping(uint256=>uint256) public totalAmountSentForPeriods;
+    uint256 public startBlock;
+    address public controller;
+    uint256 public avatarBalanceBefore;
+
+  /**
+   * @dev initialize
+   * @param _avatar the avatar to mint reputation from
+   * @param _constraintPeriodWindow the constraintPeriodWindow in blocks units
+   * @param _amountAllowedPerPeriod the amount of eth to constraint for each period
+   */
+    function initialize(
+        address _avatar,
+        uint256 _constraintPeriodWindow,
+        uint256 _amountAllowedPerPeriod,
+        address _controller
```

Comment:
why do you need the controller as a separate argument, can't you get that from the avatar_?